name: "pi20220283-app"

services:
  database:
    image: mysql:latest
    container_name: pi20220283-db
    env_file:
      - ./env/mysql.env
    networks:
      - app-network
    volumes:
      - mysql-data:/var/lib/mysql
      - ./db/init.sql:/docker-entrypoint-initdb.d/inti.sql
      - ./db/my.cnf:/etc/mysql/conf.d/my-custom.cnf:ro
    healthcheck:
      test: ["CMD", "mysqladmin", "ping","-h", "localhost", "-u","student", "--password=student"]
      interval: 10s
      timeout: 10s
      retries: 6

  backend:
    build: ./backend
    container_name: pi20220283-back
    ports:
      - "3000:3000"
    env_file:
      - ./env/backend.env
    networks:
      - app-network
    volumes:
      - ./backend:/app
      - /app/node_modules
    depends_on:
      database:
        condition: service_healthy

  frontend:
    build: ./frontend
    container_name: pi20220283-front
    ports:
      - "5000:3000"
    networks:
      - app-network
    volumes:
      - ./frontend:/app
      - /app/node_modules
    depends_on:
      - backend 
networks:
  app-network:
    driver: bridge
volumes:
  mysql-data:    
